docs/source/java/quickstartguide.rst

```diff
@@ -0,0 +1,325 @@
+    import org.apache.arrow.vector.types.pojo.Field;
+    import org.apache.arrow.vector.types.pojo.FieldType;
+    import org.apache.arrow.vector.types.pojo.Schema;
+    import java.nio.charset.StandardCharsets;
+    import java.util.HashMap;
+    import java.util.Map;
+    import static java.util.Arrays.asList;
+
+    Field a = new Field("age",
+            FieldType.nullable(new ArrowType.Int(32, true)),
+            /*children*/null
+    );
+    Field b = new Field("name",
+            new FieldType(true, new ArrowType.Utf8(), /*dictionary*/ null, /*metadata*/ null),
+            /*children*/null
+    );
+    Schema schema = new Schema(asList(a, b), /*metadata*/ null);
+    try(
+        BufferAllocator allocator = new RootAllocator();
+        VectorSchemaRoot root = VectorSchemaRoot.create(schema, allocator);
+        IntVector intVectorA = (IntVector) root.getVector("age");
+        VarCharVector varCharVectorB = (VarCharVector) root.getVector("name");
+    ){
+        root.setRowCount(3);
+        intVectorA.allocateNew(3);
+        intVectorA.set(0, 10);
+        intVectorA.set(1, 20);
+        intVectorA.set(2, 30);
+        varCharVectorB.allocateNew(3);
+        varCharVectorB.set(0, "Dave".getBytes(StandardCharsets.UTF_8));
+        varCharVectorB.set(1, "Peter".getBytes(StandardCharsets.UTF_8));
+        varCharVectorB.set(2, "Mary".getBytes(StandardCharsets.UTF_8));
+        System.out.println("VectorSchemaRoot created: \n" + root.contentToTSVString());
+    }
+
+.. code-block:: shell
+
+    VectorSchemaRoot created:
+    age	    name
+    10	    Dave
+    20	    Peter
+    30	    Mary
+
+
+Interprocess Communication (IPC)
+********************************
+
+Arrow data can be written to and read from disk, and both of these can be done in
+a streaming and/or random-access fashion depending on application requirements.
+
+**Write data to an arrow file**
+
+Example: Write the dataset from the previous example to an Arrow random-access file.
+
+.. code-block:: Java
+
+
+    import org.apache.arrow.memory.BufferAllocator;
+    import org.apache.arrow.memory.RootAllocator;
+    import org.apache.arrow.vector.IntVector;
+    import org.apache.arrow.vector.VarCharVector;
+    import org.apache.arrow.vector.VectorSchemaRoot;
+    import org.apache.arrow.vector.ipc.ArrowFileWriter;
+    import org.apache.arrow.vector.types.pojo.ArrowType;
+    import org.apache.arrow.vector.types.pojo.Field;
+    import org.apache.arrow.vector.types.pojo.FieldType;
+    import org.apache.arrow.vector.types.pojo.Schema;
+    import java.io.File;
+    import java.io.FileOutputStream;
+    import java.io.IOException;
+    import java.nio.charset.StandardCharsets;
+    import java.util.HashMap;
+    import java.util.Map;
+    import static java.util.Arrays.asList;
+
+    Map<String, String> metadataField = new HashMap<>();
+    metadataField.put("K1-Field", "K1F1");
+    metadataField.put("K2-Field", "K2F2");
```

Comment:
IMO, we can remove the metadata here to keep this example more manageable.

Reply:
Deleted